### ISSUE TITLE
FCN Accuracy in Verbose is different that when you use score[BUG]

### ISSUE THREAD

**Issue by mahi-knowlabs** [sktime/sktime#5674]
**Describe the bug**
When I train my FCN model, in the verbose, I see accuracy displaying in the 90s. When I predict it on the same training set, I get a different accuracy. Further, when I use the score method, it is different from the what I see in Verbose.

**To Reproduce**

`X_train` is a 3d np array. 
```python
clf = FCNClassifier(verbose=True, n_epochs=1000)
values = clf.predict(X_train, y_train_class)
print(clf.score(X_train))
```
What I see in verbose:
```
Epoch 999/1000
72/72 [==============================] - 1s 7ms/step - loss: 0.1999 - accuracy: 0.9126
Epoch 1000/1000
72/72 [==============================] - 1s 7ms/step - loss: 0.2283 - accuracy: 0.9012
72/72 [==============================] - 0s 2ms/step
```
What I see using score:
```
72/72 [==============================] - 0s 2ms/step
0.7159090909090909
```

**Expected behavior**

I expect both the accuracy and the score to be approximately the same. Not that much difference. 

**Additional context**

**Versions**
<details>
```

from sktime import show_versions; show_versions()
System:
    python: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
executable: /usr/bin/python3
   machine: Linux-6.1.58+-x86_64-with-glibc2.35

Python dependencies:
          pip: 23.1.2
       sktime: 0.25.0
      sklearn: 1.2.2
       skbase: 0.6.1
        numpy: 1.23.5
        scipy: 1.11.4
       pandas: 1.5.3
   matplotlib: 3.7.1
       joblib: 1.3.2
        numba: 0.58.1
  statsmodels: 0.14.1
     pmdarima: 2.0.4
statsforecast: 1.6.0
      tsfresh: 0.20.1
      tslearn: 0.5.3.2
        torch: 2.1.0+cu121
   tensorflow: 2.15.0
tensorflow_probability: 0.22.0
```
</details>

**Comment by mahi-knowlabs** [comment]
Sorry, I had a typo, This is what I meant:

```
clf = FCNClassifier(verbose=True, n_epochs=1000)
values = clf.fit(X_train, y_train_class)
print(clf.score(X_train))
```

**Comment by fkiraly** [comment]
>  initiate -> fit -> predict

`score` is a shorthand and internally calls `predict`, then sticks the result into `accuracy_score`.

The reason for the discrepancy is probably you having multiple classes, as internally the `y` is one-hot encoded, so accuracy is averaged over indicator labels, rather than computed on the multi-class problem.

In general, one-hot multioutput-binary accuracy will be higher than single-output multiclass (this is a small theorem).

Based on the numbers, I am guessing that you have four different classes in your original problem.

**Comment by mahi-knowlabs** [comment]
3 classes. That is interesting. Thank you for sharing that. I think I can modify the code as I need it.

**Comment by fkiraly** [comment]
> I think I can modify the code as I need it.

What do you need - train estimates of the classification accuracy?

Are you planning to change the printout?

**Comment by fkiraly** [comment]
> 3 classes. 

Of course - 10% (error) is a *third* of 30%, not a *fourth*.

**Comment by benHeid** [comment]
I think it is the same reason as in #5804. 

It is important to note that BatchNormalization is behaving differently during training and fitting. Below, I copy my answer on the newer issue. Thus, I would propose to close this issue. 
> The reason of this behavior are the BatchNormalization Layers. These layers behave differently during training and inference. Thus, the result printed via the training differ compared to the results calculated afterwards.
> 
> You might take a further look into:
>
> https://keras.io/api/layers/normalization_layers/batch_normalization/
>  https://github.com/keras-team/keras/issues/6977
> If you clone sktime you can test it locally also by commenting out all the BatchNormalisation Layers.

**Comment by mahi-knowlabs** [comment]
@benHeid I am not convinced the lack of performance is due to batch norm. Not enough to change performance from 90% - 70% accuracy. Something else is happening.

**Comment by mahi-knowlabs** [comment]
I will comment out batch norm and see the performance to see if you are right.

**Comment by benHeid** [comment]
Any updates?
